File: Advanced/auction-cordapp/contracts/src/test/java/net/corda/samples/auction/contracts/StateTests.java
package net.corda.samples.auction.contracts;

import net.corda.core.contracts.UniqueIdentifier;
import net.corda.samples.auction.states.AuctionState;
import net.corda.testing.node.MockServices;
import org.junit.Test;

import java.util.UUID;

public class StateTests {
    private final MockServices ledgerServices = new MockServices();

    @Test
    public void hasFieldOfCorrectType() throws NoSuchFieldException {
        // Does the message field exist?
        AuctionState.class.getDeclaredField("auctionId");
        // Is the message field of the correct type?
        assert(AuctionState.class.getDeclaredField("auctionId").getType().equals(UUID.class));
    }
}

File: kotlin-samples/Advanced/auction-cordapp/contracts/src/test/kotlin/net/corda/samples/auction/contracts/StateTests.kt
package net.corda.samples.auction.contracts

import net.corda.samples.auction.states.AuctionState
import org.junit.Test
import java.util.*
import kotlin.test.assertEquals

class StateTests {

    @Test
    fun hasFieldOfCorrectType() {
        // Does the amount field exist?
        AuctionState::class.java.getDeclaredField("auctionId")
        // Is the amount field of the correct type?
        assertEquals(AuctionState::class.java.getDeclaredField("auctionId").type, UUID::class.java)
    }
}